File: src/business/rule/engine/serviceprovider/SafetyBusinessRuleProvider.java
package business.rule.engine.serviceprovider;

import business.rule.engine.model.Order;
import business.rule.engine.service.IUserService;

public class SafetyBusinessRuleProvider implements IBusinessRuleProvider
    {
        private IUserService userService;

        public SafetyBusinessRuleProvider(IUserService userService)
        {
            this.userService = userService;
        }

        public String NameRequirement = "Safety";

        public boolean Process(Order order)
        {
            if (order.Product.GetType() == typeof(Video) &&
                order.Product?.Name == "Learning to Ski")
            {
                order.PackingSlip += "First Aid";
                return true;
            }
            return false;
        }
    }

File: src/BusinessRuleProviders/SafetyBusinessRuleProvider.cs
using Raymaker.RulesEngine.Application.Model;

namespace Raymaker.RulesEngine.Application.BusinessRuleProviders
{
    public class SafetyBusinessRuleProvider : IBusinessRuleProvider
    {
        private readonly IUserService userService;

        public SafetyBusinessRuleProvider(IUserService userService)
        {
            this.userService = userService;
        }

        public string NameRequirement => "Safety";

        // If the payment is for the video “Learning to Ski,” add a free “First Aid” video to the packing slip).
        public (bool isSatisfied, string message) Process(Order order)
        {
            if (order.Product.GetType() == typeof(Video) &&
                order.Product?.Name == "Learning to Ski")
            {
                order.PackingSlip += "First Aid";
                return (true, $"{NameRequirement}: Added First Aid video");
            }
            return (false, $"{NameRequirement}: No action");
        }
    }
}
